### PR TITLE
Improve S2437: message - replace silly with unnecessary

### PR DESCRIPTION
**Pull request by antonioaversa**
RSpec: https://github.com/SonarSource/rspec/pull/1991

### CODE REVIEW

**Comment by antonioaversa** [comment]
Uncovered lines (such as https://sonarcloud.io/code?id=sonaranalyzer-dotnet&pullRequest=7318&selected=sonaranalyzer-dotnet%3Aanalyzers%2Fsrc%2FSonarAnalyzer.CSharp%2FRules%2FUnnecessaryBitwiseOperationCodeFix.cs&line=36) are on existing and unmodified code, in scenarios where the codefix is applied to the wrong node type => not easy to reproduce in a test.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/SillyBitwiseOperationTest.cs [hunk 1]
```diff
@@ -24,41 +24,41 @@
 namespace SonarAnalyzer.UnitTest.Rules
 {
     [TestClass]
-    public class SillyBitwiseOperationTest
+    public class UnnecessaryBitwiseOperationTest
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/SillyBitwiseOperationTest.cs:27]: The name of the file is still `SillyBitwiseOperationTest`

**Review by cristian-ambrosini-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/SillyBitwiseOperationTest.cs (hunk 1 above)
Comment (antonioaversa) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/SillyBitwiseOperationTest.cs:27]: Well spot, fixed!

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**